caffe2/operators/utility_ops.h

```diff
@@ -315,6 +356,59 @@ class SumOp : public Operator<Context> {
   }
 };
 
+template <class Context>
+class SumGradientOp : public Operator<Context> {
+ public:
+  USE_OPERATOR_CONTEXT_FUNCTIONS;
+  USE_SIMPLE_CTOR_DTOR(SumGradientOp);
+
+  template <typename DstType>
+  bool DoRunWithType() {
+    CAFFE_ENFORCE_GT(InputSize(), 1);
+    CAFFE_ENFORCE_EQ(OutputSize(), InputSize() - 1);
+
+    const auto& dY = Input(0);
+    std::vector<int> dY_dims(dY.sizes().cbegin(), dY.sizes().cend());
+
+    for (auto i = 1; i < InputSize(); i++) {
+      const auto& X = Input(i);
+      std::vector<int> X_dims(X.sizes().cbegin(), X.sizes().cend());
+
+      std::vector<int> dY_dims_backward, X_dims_backward;
+      elementwise_ops_utils::ComputeBinaryBroadcastBackwardDims(
+          dY_dims, X_dims, &dY_dims_backward, &X_dims_backward);
+
+      auto* dX = Output(i - 1, X.sizes(), at::dtype<DstType>());
+
+      math::ReduceSum(
+          dY_dims.size(),
+          dY_dims.data(),
+          X_dims_backward.data(),
+          DstType(1),
+          dY.template data<DstType>(),
+          dX->template mutable_data<DstType>(),
+          &context_);
+    }
+
+    return true;
+  }
+
+  bool RunOnDevice() override {
+    CAFFE_ENFORCE_GT(InputSize(), 1);
+
+    if (Input(1).template IsType<float>()) {
+      return DoRunWithType<float>();
+    } else if (Input(1).template IsType<int>()) {
```

Comment:
The original version support int or not?

Reply:
In original version, SumOp support float and int in CPU. So perhaps we stick with that?